### PR TITLE
chore(sqllab): Typescript for SqlEditor component

### PR DESCRIPTION
**Pull request by justinpark**
### SUMMARY
This commit migrates SqlEditor jsx to tsx along with all associated type updates.

### TESTING INSTRUCTIONS
npm run type

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/AceEditorWrapper/index.tsx [hunk 1]
```diff
@@ -30,7 +31,7 @@ type HotKey = {
   key: string;
   descr: string;
   name: string;
-  func: () => void;
+  func: (arg0: IAceEditor) => void;
```
Comment (michael-s-molina) [inline superset-frontend/src/SqlLab/components/AceEditorWrapper/index.tsx:34]:
```suggestion
  func: (aceEditor: IAceEditor) => void;
```

File: superset-frontend/src/SqlLab/components/SaveQuery/index.tsx [hunk 2]
```diff
@@ -38,11 +39,11 @@ interface SaveQueryProps {
   columns: ISaveableDatasource['columns'];
   onSave: (arg0: QueryPayload, id: string) => void;
   onUpdate: (arg0: QueryPayload, id: string) => void;
-  saveQueryWarning: string | null;
-  database: Record<string, any>;
+  saveQueryWarning?: string | null;
```
Comment (michael-s-molina) [inline superset-frontend/src/SqlLab/components/SaveQuery/index.tsx:42]: Can this be `undefined` or `string | null` is sufficient?

File: superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx [hunk 3]
```diff
@@ -55,15 +55,15 @@ import {
...
+export interface ExtendedTable extends Table {
   expanded: boolean;
 }
 
 interface SqlEditorLeftBarProps {
   queryEditorId: string;
   height?: number;
   tables?: ExtendedTable[];
-  database: DatabaseObject;
+  database: DatabaseObject | undefined;
```
Comment (michael-s-molina) [inline superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx:66]:
```suggestion
  database?: DatabaseObject;
```

**Review by michael-s-molina — COMMENTED** [review]
Thank you for the PR @justinpark. I left some first-pass comments.

File: superset-frontend/src/SqlLab/components/SqlEditor/index.tsx [hunk 4]
```diff
@@ -471,10 +494,11 @@ const SqlEditor = ({
   const onResizeStart = () => {
     // Set the heights on the ace editor and the ace content area after drag starts
     // to smooth out the visual transition to the new heights when drag ends
+    // @ts-ignore
     document.getElementsByClassName('ace_content')[0].style.height = '100%';
```
Comment (michael-s-molina) [inline superset-frontend/src/SqlLab/components/SqlEditor/index.tsx:498]:
```suggestion
    (
      document.getElementsByClassName('ace_content')[0] as HTMLElement
    ).style.height = '100%';
```

File: superset-frontend/src/SqlLab/components/SqlEditor/index.tsx [hunk 5]
```diff
@@ -566,6 +584,7 @@ const SqlEditor = ({
           <AntdSwitch
             checked={autocompleteEnabled}
             onChange={handleToggleAutocompleteEnabled}
+            // @ts-ignore
```
Comment (michael-s-molina) [inline superset-frontend/src/SqlLab/components/SqlEditor/index.tsx:587]: This property does not exist. I think you can safely remove it as it's not being referenced anywhere. I searched for "autocomplete-switch" and I wasn't able to find anything.

**Comment by justinpark** [comment]
@michael-s-molina could you review the update again?

**Review by michael-s-molina — APPROVED** [review]
Code LGTM. Thanks for addressing all comments @justinpark!

**Pull request merged**